pilot/pkg/networking/plugin/authz/authorization.go

```diff
@@ -47,17 +47,37 @@ func NewBuilder(actionType ActionType, push *model.PushContext, proxy *model.Pro
 }
 
 func NewBuilderForService(actionType ActionType, push *model.PushContext, proxy *model.Proxy, useFilterState bool, svc *model.Service) *Builder {
+	return newBuilder(actionType, push, proxy, useFilterState, svc, false)
+}
+
+// NewWaypointTerminationBuilder creates a builder for use on the waypoints HBONE termination layer
+func NewWaypointTerminationBuilder(actionType ActionType, push *model.PushContext, proxy *model.Proxy) *Builder {
+	return newBuilder(actionType, push, proxy, false, nil, true)
+}
+
+func newBuilder(actionType ActionType, push *model.PushContext, proxy *model.Proxy, useFilterState bool, svc *model.Service, alwaysTreatAsNonWaypoint bool) *Builder {
 	tdBundle := trustdomain.NewBundle(push.Mesh.TrustDomain, push.Mesh.TrustDomainAliases)
 	option := builder.Option{
 		IsCustomBuilder: actionType == Custom,
 		UseFilterState:  useFilterState,
 	}
 	selectionOpts := model.PolicyMatcherForProxy(proxy).WithService(svc)
+	if alwaysTreatAsNonWaypoint {
+		selectionOpts.IsWaypoint = false
+	}
```

Comment:
The purpose here is to build authz rules for a waypoint, but use selection semantics of a "regular" workload to allow it to pick up policy which would normally be applied against workloads by ztunnel. I think outside the context of the PR that might be lost so renaming the variable for clarity and/or a comment explaining this would probably be helpful.